src/mixins/VirtualScrolling.js

```diff
@@ -0,0 +1,153 @@
+/**
+ * @copyright Copyright (c) 2022 Louis Chemineau <louis@chmn.me>
+ *
+ * @author Louis Chemineau <louis@chmn.me>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * VirtualScrolling component to limit the number of elements present in the DOM.
+ * Usage template:
+ *     <template>
+ *         <div ref="vsContainer"
+ *             <div ref="vsItemsContainer"
+ *                 :style="vsItemsContainerStyle">
+ *                 <div v-for="row of vsVisibleRows"
+ *                     :key="row.key"
+ *                     :style="{height: `${row.height}px`}">
+ *                     <div v-for="item of row.items"
+ *                         :key="item.id">
+ *                         <!-- Render your component here -->
+ *                     </div>
+ *                 </div>
+ *             </div>
+ *         </div>
+ *     </template>
+ *     <script>
+ *         export default {
+ *             mixins: [VirtualScrolling],
+ *             computed: {
+ *                 rows() {
+ *                     // Return your rows here.
+ *                     return []
+ *                 },
+ *             },
+ *         }
+ *     </script>
+ *     <style>
+ *         .vs-container {
+ *             height: 500px; // Static height
+ *             overflow-y: scroll;
+ *         }
+ *     </style>
+ */
```

Comment:
All of this should most likely be a direct component with a render function. We would gain a lot of performances :)